clippy_lints/src/utils/hir_utils.rs

```diff
@@ -574,4 +581,99 @@ impl<'a, 'tcx> SpanlessHash<'a, 'tcx> {
             },
         }
     }
+
+    pub fn hash_lifetime(&mut self, lifetime: &Lifetime) {
```

Comment:
```suggestion
    pub fn hash_lifetime(&mut self, lifetime: &Lifetime) {
        std::mem::discriminant(&lifetime.name).hash(&mut self.s);
```

Reply:
I also hashed the discriminant for the name type inside `Lifetime::Param` if that's okay?